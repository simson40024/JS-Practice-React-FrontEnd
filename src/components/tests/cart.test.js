import Cart from "../cart/cart";
import { createRoot } from "react-dom/client";
import { act } from "@testing-library/react";
import * as hooks from "../../hook";

let container = null;
beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  container.remove();
  container = null;
});

const state = {
  user: {
    value: {
      name: "",
      token: "",
    },
  },
  currency: {
    currency: "UAH",
    rate: 38,
  },
  cart: {
    products: [
      {
        id: "12345",
        name: "cool t-shot",
        quantity: 2,
        price: 20,
        imgUrl: "./img1.jpg",
      },
      {
        id: "67890",
        name: "cool dress",
        quantity: 1,
        price: 50,
        imgUrl: "./img1.jpg",
      },
    ],
  },
};

const mock = jest.spyOn(hooks, "useAppSelector");
const mockDispatch = jest.spyOn(hooks, "useAppDispatch");

describe("Cart", () => {
  it("should create Cart", () => {
    mock.mockImplementation((selector) => selector(state));
    mockDispatch.mockReturnValue(jest.fn());
    const total = 150;
    act(() => {
      createRoot(container).render(<Cart />);
    });
    expect(container).toMatchSnapshot();
  });
});
