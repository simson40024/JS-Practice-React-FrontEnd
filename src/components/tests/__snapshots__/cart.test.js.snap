// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should create Cart 1`] = `
<div>
  <div
    class="cart-header"
  >
    <div
      class="cart-product"
    >
      <p>
        Product
      </p>
    </div>
    <div
      class="cart-price"
    >
      <p>
        Price
      </p>
    </div>
    <div
      class="cart-quantity"
    >
      <p>
        Quantity
      </p>
    </div>
    <div
      class="cart-total"
    >
      <p>
        Total
      </p>
    </div>
    <div
      class="cart-button"
    />
  </div>
  <hr />
  <div
    class="cart-row"
    id="12345"
  >
    <div
      class="cart-product"
    >
      <img
        alt=""
        src="./img1.jpg"
        width="150px"
      />
      <p>
        cool t-shot
      </p>
    </div>
    <div
      class="cart-price"
    >
      <p>
        760
         
        <br
          class="hidden-br"
        />
        UAH
      </p>
    </div>
    <div
      class="product-quantity"
    >
      <button
        role="button"
      >
        -
      </button>
      <input
        readonly=""
        type="text"
        value="2"
      />
      <button
        role="button"
      >
        +
      </button>
    </div>
    <div
      class="cart-total"
    >
      <p>
        1520
         
        <br
          class="hidden-br"
        />
        UAH
      </p>
    </div>
    <div
      class="cart-button"
    >
      <button
        title="delete produrt"
      >
        X
      </button>
    </div>
  </div>
  <div
    class="cart-row"
    id="67890"
  >
    <div
      class="cart-product"
    >
      <img
        alt=""
        src="./img1.jpg"
        width="150px"
      />
      <p>
        cool dress
      </p>
    </div>
    <div
      class="cart-price"
    >
      <p>
        1900
         
        <br
          class="hidden-br"
        />
        UAH
      </p>
    </div>
    <div
      class="product-quantity"
    >
      <button
        role="button"
      >
        -
      </button>
      <input
        readonly=""
        type="text"
        value="1"
      />
      <button
        role="button"
      >
        +
      </button>
    </div>
    <div
      class="cart-total"
    >
      <p>
        1900
         
        <br
          class="hidden-br"
        />
        UAH
      </p>
    </div>
    <div
      class="cart-button"
    >
      <button
        title="delete produrt"
      >
        X
      </button>
    </div>
  </div>
  <hr />
  <div
    class="cart-amount"
  >
    <button>
      Confirm order
    </button>
    <p
      class="amount"
    >
      3420
       
      UAH
    </p>
  </div>
</div>
`;
